Yosemite/YosemiteTests/Mocks/MockPointOfSalePurchasableItemFetchStrategy.swift

```diff
@@ -5,10 +5,17 @@ import Foundation
 final class MockPointOfSalePurchasableItemFetchStrategy: PointOfSalePurchasableItemFetchStrategy {
     var fetchProductsCalled = false
     var spyFetchProductsPageNumber: Int?
+    var mockPagedProducts: PagedItems<POSProduct>?
+    var mockPagedVariations: PagedItems<ProductVariation>?
+    var mockError: Error?
```

Comment:
nit: how about 2 separate mock variables with a result type like `Result<PagedItems<POSProduct>?, Error>` / `Result<PagedItems< ProductVariation>?, Error>` for the 2 methods? Might look prettier, and the current shared error might not work if we want one function to return an error but not the other one.

Reply:
Yeah, I'll give that a go – I kind of dismissed it because it probably makes all the call sites slightly more complex, but it does make the mock more flexible.

Reply:
[7153940](https://github.com/woocommerce/woocommerce-ios/pull/15710/commits/715394067c63d948ff7b55897daa486945cb8cdd)